test/jdk/javax/xml/jaxp/datatype/8033980/SerializationTest.java

```diff
@@ -25,160 +25,295 @@
-            XMLGregorianCalendar c = dtf.newXMLGregorianCalendar(EXPECTED_CAL);
-            Duration d = dtf.newDuration(EXPECTED_DURATION);
-            toFile((Serializable) c, filePath + javaVersion + FILENAME_CAL);
-            toFile((Serializable) d, filePath + javaVersion + FILENAME_DURATION);
-        } catch (Exception e) {
-            fail(e.getMessage());
-        }
+    @DataProvider(name = "GregorianCalendarData")
+    public Object [][] gregorianCalendarDataBytes() {
+        return new Object [][] {{System.getProperty("java.version"), gregorianCalendarAndDurationSerData[0], EXPECTED_CAL},
+                {JDK[0], gregorianCalendarAndDurationSerData[1], EXPECTED_CAL},
+                {JDK[1], gregorianCalendarAndDurationSerData[2], EXPECTED_CAL},
+                {JDK[2], gregorianCalendarAndDurationSerData[3], EXPECTED_CAL},
+                {JDK[3], gregorianCalendarAndDurationSerData[4], EXPECTED_CAL}};
     }
 
     /**
-     * Read the object from a file.
+     *Provide data for JDK version and serialized Gregorian Calendar Base64 encoded string
      */
-    private static Object fromFile(String filePath) throws IOException,
+    @DataProvider(name = "GregorianCalendarDataBase64")
+    public Object [][] gregorianCalendarDataBase64() {
+        return new Object [][] {{System.getProperty("java.version"), gregorianCalendarAndDurationSerData[0], EXPECTED_CAL},
+                {JDK[2], gregorianCalendarAndDurationSerData[3], EXPECTED_CAL},
+                {JDK[3], gregorianCalendarAndDurationSerData[4], EXPECTED_CAL}};
+    }
+
+    /**
+     *Provide data for JDK version and Duration serialized bytes
+     */
+    @DataProvider(name = "DurationData")
+    public Object [][] DurationData() {
+        return new Object [][] {{System.getProperty("java.version"), gregorianCalendarAndDurationSerData[0], EXPECTED_DURATION},
+                {JDK[0], gregorianCalendarAndDurationSerData[1], EXPECTED_DURATION},
+                {JDK[1], gregorianCalendarAndDurationSerData[2], EXPECTED_DURATION},
+                {JDK[2], gregorianCalendarAndDurationSerData[3], EXPECTED_DURATION},
+                {JDK[3], gregorianCalendarAndDurationSerData[4], EXPECTED_DURATION}};
+    }
+
+    /**
+     *Provide data for JDK version and Duration Base64 encode serialized bytes string
+     */
+    @DataProvider(name = "DurationDataBase64")
+    public Object [][] DurationDataBase64() {
+        return new Object [][] {{System.getProperty("java.version"), gregorianCalendarAndDurationSerData[0], EXPECTED_DURATION},
+                {JDK[2], gregorianCalendarAndDurationSerData[3], EXPECTED_DURATION},
+                {JDK[3], gregorianCalendarAndDurationSerData[4], EXPECTED_DURATION}};
+    }
+
+    /**
+     * verify serialization compatibility for XMLGregorianCalendar
+     * @param javaVersion
+     * @param gcsd
+     * @param gregorianDate
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
```

Comment:
See above regarding using a block comment or using the proper javadoc comment format.   Either way, please define what the param values are used for